environments/tangle/src/api.rs

```diff
@@ -1,131 +1,72 @@
-                AccountId32,
-                u64,
-                MaxParticipants,
-                MaxKeyLen,
-                MaxDataLen,
-                MaxSignatureLen,
-                MaxSubmissionLen,
-                MaxProofLen,
-                MaxAdditionalParamsLen,
-            >,
-        >,
-        gadget_common::Error,
-    >;
+/// A client for interacting with the services API
+/// TODO: Make this the object that is used for getting services and submitting transactions as necessary.
+pub struct ServicesClient<C: Config> {
+    rpc_client: OnlineClient<C>,
+    logger: DebugLogger,
+}
 
-    /// Queries next job ID.
-    ///
-    ///  # Returns
-    ///  Next job ID.
-    async fn query_next_job_id(&self, at: [u8; 32]) -> Result<u64, gadget_common::Error>;
+impl<C: Config> ServicesClient<C> {
+    pub fn new(logger: DebugLogger, rpc_client: OnlineClient<C>) -> Self {
+        Self { logger, rpc_client }
+    }
 
-    /// Queries restaker's role key
-    ///
-    ///  # Returns
-    ///  Role key
-    async fn query_restaker_role_key(
-        &self,
-        at: [u8; 32],
-        address: AccountId32,
-    ) -> Result<Option<Vec<u8>>, gadget_common::Error>;
+    pub fn rpc_client(&self) -> &OnlineClient<C> {
+        &self.rpc_client
+    }
+}
 
-    /// Queries restaker's roles
-    ///
-    /// # Returns
-    /// List of roles enabled for restaker
-    async fn query_restaker_roles(
+impl<C: Config> ServicesClient<C>
+where
+    BlockRef<<C as Config>::Hash>: From<BlockRef<H256>>,
+{
+    pub async fn query_restaker_blueprints(
         &self,
         at: [u8; 32],
         address: AccountId32,
-    ) -> Result<Vec<tangle_runtime::RoleType>, gadget_common::Error>;
+    ) -> Result<Vec<ServiceBlueprint>, gadget_common::Error> {
+        let block_ref = BlockRef::from_hash(H256::from_slice(&at));
+        let call = api::storage().services().user_services(address);
+        let blueprint_ids: Vec<u64> = self
+            .rpc_client
+            .storage()
+            .at(block_ref.clone())
+            .fetch_or_default(&call)
+            .await
+            .map(|v| v.0)
+            .map_err(|e| gadget_common::Error::ClientError { err: e.to_string() })?;
+
+        let mut ret = vec![];
+
+        for blueprint_id in blueprint_ids {
+            let svcs = api::storage().services().blueprints(blueprint_id);
+            let blueprint = self
+                .rpc_client
+                .storage()
+                .at(block_ref.clone())
+                .fetch(&svcs)
+                .await
+                .map_err(|e| gadget_common::Error::ClientError { err: e.to_string() })?;
+
+            if let Some(blueprint) = blueprint {
+                ret.push(blueprint.1);
+            } else {
+                self.logger.error(format!(
+                    "Failed to fetch blueprint for ID: {}",
+                    blueprint_id
+                ));
+            }
+        }
+
+        Ok(ret)
```

Comment:
This implementation does not seem to be correct. It queries the user services', not the operator's. We should rename this method to be more consistent `services_with_blueprints_by_operator` and use the RPC method provided for you.
https://github.com/webb-tools/tangle/blob/5384f0d19664ab7332a4c191e745c326f0416218/tangle-subxt/src/tangle_testnet_runtime.rs#L686